### ISSUE TITLE
Providers must only be specified either at the TestCase or TestStep level

### ISSUE THREAD

**Issue by mschuchard** [hashicorp/terraform-plugin-testing#176]
### terraform-plugin-testing version
```
1.5.0
```

### Relevant provider source code

```go
resource.ParallelTest(test, resource.TestCase{
		PreCheck:                 func() { provider.AccPreCheck(test) },
		ProtoV6ProviderFactories: provider.AccProtoV6ProviderFactories,
		Steps: []resource.TestStep{
			{
				Config: provider.ProviderConfig + fmt.Sprintf(datasourceConfig, foo),
				Check: resource.ComposeAggregateTestCheckFunc(
                                ...
				),
			},
		},
	})
```

### Terraform Configuration Files
n/a

### Expected Behavior
The acceptance test case should invoke and execute as expected which is what occurs in version 1.4.0. I have checked the CHANGELOG for 1.5.0 and see nothing about this backwards-incompatible change.

If this issue derives from documentation and not code, then please update CHANGELOG etc. for the change, and clarify the error message in the code as the provider is being specified at the `TestCase` level according to the above code.

### Actual Behavior
> Test validation error: TestStep 1/1 validation error: Providers must only be specified either at the TestCase or TestStep level

### Steps to Reproduce
`TF_ACC=1 go test -v ./<provider>/...`

### References
not according to search

**Comment by bflad** [comment]
Hi @mschuchard 👋 Thank you for raising this issue and apologies for this regression. It certainly was not intended and I'm guessing there was an existing unit test to cover this valid behavior. The team is investigating and will release a patch update to reinstate the prior behavior per our versioning and compatibility promises.

**Comment by mschuchard** [comment]
Thanks; I have downgraded to 1.4.0 in the interim as that still enables `PlanCheck` which is the most recent feature I am currently utilizing, so no rush.

**Comment by bendbennett** [comment]
@mschuchard I believe the regression you reported should be resolved in `v1.5.1`. Please let us know by opening an issue if you encounter any other problems. Thanks for raising the issue.

**Comment by mschuchard** [comment]
This also helps me with understanding root cause, and why another of my plugins does not have this issue. The other provider can be completely configured with `env`, but this one cannot (ergo why AT004 is flagged during linting). Therefore I must also include a stub in `Config`.
Thanks!
